### PR TITLE
add solaris support

### PR DESCRIPTION
**Pull request by chris-rock**
- support for solaris 10 & 11
- also tested against SmartOS and OmniOS
- improves platform detection
- optimize stats handling for Solaris 10

### CODE REVIEW

File: lib/train/extras/file_solaris.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# encoding: utf-8
+# author: Dominik Richter
+# author: Christoph Hartmann
+
+require 'shellwords'
```
Comment (srenatus) [inline lib/train/extras/file_solaris.rb]: Go shellwords! ...but it's not actually used, is it?

File: lib/train/extras/file_solaris.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+require 'train/extras/stat'
+
+module Train::Extras
+  class SolarisFile < LinuxFile
+    def initialize(backend, path)
+      super(backend, path)
```
Comment (srenatus) [inline lib/train/extras/file_solaris.rb]: passing the arguments as-is is the default btw, so `super` would be sufficient

File: lib/train/extras/file_solaris.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    end
+
+    def content
+      return @content if defined?(@content)
+      @content = case
```
Comment (srenatus) [inline lib/train/extras/file_solaris.rb]: line 15 and 16 could be combined to

```
@content ||= case ...
```

**Comment by srenatus** [comment]
Sorry for nitpicking :wink:

**Comment by chris-rock** [comment]
@srenatus +1 love it

**Comment by srenatus** [comment]
:grinning:

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @arlimus

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Pull request merged**